images/virtualization-artifact/pkg/controller/moduleconfig/reduce_cidrs_validator.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moduleconfig
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/deckhouse/virtualization-controller/pkg/common/ip"
+	mcapi "github.com/deckhouse/virtualization-controller/pkg/controller/moduleconfig/api"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+const virtualMachineCIDRs = "virtualMachineCIDRs"
+
+type reduceCIDRsValidator struct {
```

Comment:
Thoughts on naming. This validation is not about "reducing" CIDRs items, it is about "removing" used CIDRs. I think of this case: new CIDR added and used CIDR is removed, no changes in CIDRs length, nothing reduced, but _removed_ CIDR should be validated.